ocaml/xapi/test_xapi_xenops.ml

```diff
@@ -29,6 +29,43 @@ let test_enabled_in_xenguest () =
       let e = val_fn (p x) in
       if e then err x) should_be_false
 
+let simulator_setup = ref false
+let setup_simulator () =
+  if not !simulator_setup then begin
+    Printf.printf "XXX Starting simulator!!!\n%!";
+    Xenops_utils.set_fs_backend (Some (module Xenops_utils.MemFS: Xenops_utils.FS));
+    Xenops_server.register_objects ();
+    Xenops_server.set_backend (Some (module Xenops_server_simulator: Xenops_server_plugin.S));
+    Scheduler.start ();
+    Xenops_server.WorkerPool.start 16;
+    Xcp_client.use_switch := true;
+    Xapi_xenops_queue.queue_override := ["simulator",Xenopsd.rpc_fn];
+    simulator_setup := true;
+  end else ()
+
+let test_vm_start () =
+  let __context = make_test_database () in
+  setup_simulator ();
+  Helpers.test_mode := true;
+  let open Xenops_interface in
+  let open Xapi_xenops_queue in
+  let module Client = (val make_client "simulator" : XENOPS) in
+  let _ = Client.VM.list "dbg" in
+  let _ = Thread.create Xapi_xenops.events_from_xenopsd "simulator" in
+  let vm = make_vm ~__context () in
+  let flags = [
+    Xapi_globs.cpu_info_vendor_key, "AuthenticAMD";
+    Xapi_globs.cpu_info_features_key, "deadbeef-deadbeef";
+  ] in
+  Db.VM.set_last_boot_CPU_flags ~__context ~self:vm ~value:flags;
+  Db.VM.add_to_other_config ~__context ~self:vm ~key:"xenops" ~value:"simulator";
+  try
+    Xapi_xenops.start ~__context ~self:vm false false
```

Comment:
I thought this was a neat way to test more of xapi's code - we can do more simulator libraries for the other daemons and use them in the same way :-)